core/src/test/java/org/apache/iceberg/TestWapWorkflow.java

```diff
@@ -250,7 +254,12 @@ public void testRollbackAndCherrypick() {
     Snapshot thirdSnapshot = base.currentSnapshot();
 
     // rollback to first snapshot
-    table.manageSnapshots().rollbackTo(firstSnapshotId).commit();
+    ManageSnapshots manageSnapshots = table.manageSnapshots().rollbackTo(firstSnapshotId);
+    if (manageSnapshots instanceof SnapshotManager) {
```

Comment:
why is this `instanceof` check needed? `rollbackTo` returns only one implementation of rollback. `RollbackToSnapshot` was only kept for backwards compatibility of the the api but it internally adapts to `SnapshotManager` impl.